src/uct/ib/dc/dc_mlx5.c

```diff
@@ -885,17 +885,45 @@ int uct_dc_mlx5_iface_is_reachable(const uct_iface_h tl_iface,
                                    const uct_device_addr_t *dev_addr,
                                    const uct_iface_addr_t *iface_addr)
 {
-    uct_dc_mlx5_iface_addr_t *addr = (uct_dc_mlx5_iface_addr_t *)iface_addr;
+    const uct_iface_is_reachable_params_t params = {
+        .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
+        .device_addr        = dev_addr,
+        .iface_addr         = iface_addr,
+        .info_string        = NULL,
+        .info_string_length = 0
+    };
+    
+    return uct_dc_mlx5_iface_is_reachable_v2(tl_iface, &params);
+}
+
+int uct_dc_mlx5_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                      const uct_iface_is_reachable_params_t *params)
+{
+    const uct_iface_addr_t *iface_addr = params->iface_addr;
+    char *info_string                  = params->info_string;
+    size_t info_string_length          = params->info_string_length;
+    uct_dc_mlx5_iface_addr_t *addr     = (uct_dc_mlx5_iface_addr_t *)iface_addr;
     uct_dc_mlx5_iface_t UCS_V_UNUSED *iface;
 
     iface = ucs_derived_of(tl_iface, uct_dc_mlx5_iface_t);
     ucs_assert_always(iface_addr != NULL);
 
-    return ((addr->flags & UCT_DC_MLX5_IFACE_ADDR_DC_VERS) ==
-            iface->version_flag) &&
-           (UCT_DC_MLX5_IFACE_ADDR_TM_ENABLED(addr) ==
-            UCT_RC_MLX5_TM_ENABLED(&iface->super)) &&
-           uct_ib_iface_is_reachable(tl_iface, dev_addr, iface_addr);
+    if ((addr->flags & UCT_DC_MLX5_IFACE_ADDR_DC_VERS) != iface->version_flag) {
+        snprintf(info_string, info_string_length,
+                 "iface %p: unreachable due to DC version mismatch", iface);
+        return 0;
+    }
+
+    if (UCT_DC_MLX5_IFACE_ADDR_TM_ENABLED(addr) !=
+        UCT_RC_MLX5_TM_ENABLED(&iface->super)) {
+        snprintf(info_string, info_string_length,
+                 "iface %p: unreachable due to TM mismatch", iface);
```

Comment:
`return 0;`

Reply:
Does you mean that add empty line before `return 0;`?

Reply:
Got it. Thanks